lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -0,0 +1,39 @@
+# Code-related anti-patterns
+
+This document outlines anti-patterns related to your code and particular Elixir idioms and features.
+
+## Comments
+
+TODO.
+
+## Long parameter list
+
+TODO.
+
+## Complex branching
+
+TODO.
+
+## Complex else clauses in with
+
+TODO.
+
+## Complex extractions in clauses
+
+TODO.
+
+## Dynamic map fields access
+
+TODO.
+
+## Dynamic atom creation
+
+TODO.
+
+## Namespace violation
+
+TODO. (note: this is defining modules not in your namespace)
```

Comment:
@lucasvegi this is a new anti-pattern, defined here: https://github.com/elixir-lang/elixir/blob/v1.15/lib/elixir/pages/library-guidelines.md#avoid-defining-modules-that-are-not-in-your-namespace

Any suggestions on the name?

Reply:
I wonder if the name "violation" can be too... "violent"? I wanted to propose some maybe "kinder" alternatives 😊 

* *Namespace breach*
* *Namespace infringement*

Thoughts?

Reply:
+1 for namespace infringement. It has a more "trespassing" vibe for me.

Reply:
Fuck, "trespassing" is awesome 😄 I vote for *Namespace trespassing* :100

Reply:
Just thought about that and came here to edit my comment, I too vote for "namespace trespassing" :laughing:

Reply:
> https://github.com/elixir-lang/elixir/blob/v1.15/lib/elixir/pages/library-guidelines.md#avoid-defining-modules-that-are-not-in-your-namespace

@josevalim I found the suggested name "Namespace trespassing" fantastic! However, this smell seems to be practically the same as the one we already cataloged at: [Modules with identical names](https://github.com/lucasvegi/Elixir-Code-Smells#modules-with-identical-names). Am I correct?

Reply:
Not really. Duplicate modules already warn, so there is no reason to put it as a code smell. 😊

Reply:
@lucasvegi Identical names happens when the name is the same. This is about defining something in someone's else namespace. The confusion may happen because in other languages, the namespace is entity, so you cannot clobber them (to clobber, you would have to define something with identical name). In Elixir, that's not the case! A namespace is a convention, so you can define something inside a namespace without defining the namespace itself.

Reply:
> Modules with identical names

@whatyouhide I still believe we are talking about the same thing 😄 . 
Please take a look at the content of the [link](https://github.com/lucasvegi/Elixir-Code-Smells#modules-with-identical-names) I mentioned from the catalog of Elixir smells and the code examples. This smell was even extracted from the same [link](https://github.com/elixir-lang/elixir/blob/v1.15/lib/elixir/pages/library-guidelines.md#avoid-defining-modules-that-are-not-in-your-namespace) in the Elixir documentation mentioned by @josevalim. 

Please note that I really liked the new suggested name, and I see no problem in using it. I just wanted to confirm if it refers to the same smell. This would be interesting because "Modules with identical names" has already been validated in our scientific research involving 182 Elixir developers. So in the official documentation, for example, we could mention that it is the same smell with a renamed title, thus preserving its previously established history.

For all smells in our catalog, we validate their levels of relevance (negative impacts) and prevalence. These levels have practical implications for prioritizing the removal or prevention of each smell. That's why I didn't want to take the risk of losing the history of a validated smell.

Reply:
@lucasvegi in my opinion, the smell "Modules with identical names" does not accurately portray the scenario described in "Avoid defining modules that are not in your "namespace"". We are not concerned with identical names because the Elixir compiler already warns about it and a project configured with `--warnings-as-errors` wouldn't even compile (which is why it was removed). So, from my point of view, the namespace issue was not part of the original catalogue (it seems it got lost in translation!). :)

Reply:
> This would be interesting because "Modules with identical names" has already been validated in our scientific research involving 182 Elixir developers

I don't want to dwell on this too much, but I have not had the time to read the paper yet, sorry about that! I'm curious to know what the "validation" here was, I'd be super surprised if 182 Elixir developers were out redefining modules 🙈

Reply:
> > This would be interesting because "Modules with identical names" has already been validated in our scientific research involving 182 Elixir developers
> 
> I don't want to dwell on this too much, but I have not had the time to read the paper yet, sorry about that! I'm curious to know what the "validation" here was, I'd be super surprised if 182 Elixir developers were out redefining modules 🙈

Indeed, it is one of the least prevalent smells in the catalog, according to developers' perception. However, its level of relevance (negative impact) was moderate. In other words, from the developers' perspective, this smell is rare but should not be ignored when it occurs. :)